### PR TITLE
Flesh out sql/doc.go based on docs/structured.md.

### PR DESCRIPTION
**Pull request by petermattis**

### CODE REVIEW

**Comment by petermattis** [comment]
This ended up not being a straight movement of the contents of `docs/structured.md` as some of that info felt dated. I'm happy to re-add info that others feel is crucial. For example, perhaps I should have a section describing how the "system.namespace" schema allows for efficient renaming of databases/tables.

**Comment by MBerhault** [comment]
LGTM. I wouldn't go too crazy for now, a lot of this stuff is still massively in flux.

File: sql/doc.go [hunk 1]
```diff
@@ -13,453 +13,237 @@
...
+key-value map. The sql package builds on top of this core system adding
+parsing, query planning and query execution as well as defining the privilege
+model.
+
+Databases and Tables
+
+The two primary objects are databases and tables. A database is a namespace
+which holds a series of tables. Conceptually, a database can be viewed like a
+directory in a filesystem plus some additional metadata (privileges). A table
+is like a file on steriods: containing a structured layout of rows and columns
```
Comment (tamird) [inline sql/doc.go]: steroids

**Comment by tamird** [comment]
lgtm

File: sql/doc.go (hunk 1 above)
Comment (petermattis) [inline sql/doc.go]: Fixed.

**Pull request merged**